torch/csrc/jit/python/script_init.cpp

```diff
@@ -710,6 +710,22 @@ void extra_files_to_python(const ExtraFilesMap& m, const py::dict& pydict) {
   }
 }
 
+void compilationUnit_py_define(
+    CompilationUnit& cu,
+    const std::string& src,
+    const ResolutionCallback* rcb,
+    const uint32_t _frames_up) {
+  if (rcb && *rcb) {
+    cu.define(c10::nullopt, src, pythonResolver(*rcb), nullptr);
+  } else {
+    py::object py_default_rcb =
+        py::module_::import("torch._jit_internal")
+            .attr("createResolutionCallbackFromFrame")(_frames_up);
```

Comment:
Yeah, so it works because it is the Python stack that is relevant and it's separate from the C++ stack
To me, it looks like the current split is mostly historically grown. Back in the day, `torch.jit.CompilationUnit` was a Python thing that provided its functionality through `_C.ScriptModule`s. When #19720 exposed `CompilationUnit` from C++ via PyBind, this part was not cleaned up, apparently.